### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by florencegogovor**

### CODE REVIEW

**Comment by florencegogovor** [comment]
All methods for homework 2

File: FromZeroToHero.sln [hunk 1]
```diff
@@ -1,15 +1,177 @@
 ﻿
-Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (westonpace) [inline FromZeroToHero.sln:2]: It looks like you committed your homework as a file named `FromZeroToHero.sln`.  It's a little confusing.

File: FromZeroToHero.sln [hunk 2]
```diff
@@ -1,15 +1,177 @@
...
-EndGlobal
+using System;
+
+namespace Homework2
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            int[] numbers = new int[] { 10, 5, 9, 87, 56 };
```
Comment (westonpace) [inline FromZeroToHero.sln:10]: This is good for personal testing but when submitting try and avoid having a bunch of commented lines like this.  Maybe you could have them all enabled.

File: FromZeroToHero.sln [hunk 3]
```diff
@@ -1,15 +1,177 @@
...
+            // AddElementAtAnyPosition(numbers, 33, 4);
+            //RemoveElementAtStart(numbers);
+            //RemoveElementAtEnd(numbers);
+            // RemoveElementAtPositon(numbers, 3);
+            LoginInfo("KIN", "21lome");
+
+
+        }
+
+        private static void SortArray(int[] numbers)
```
Comment (westonpace) [inline FromZeroToHero.sln:23]: Interesting choice on making the methods `private`.  We haven't gone over this concept yet but I would say it is appropriate here.

File: FromZeroToHero.sln [hunk 4]
```diff
@@ -1,15 +1,177 @@
...
+        {
+            int temp = 0;
+            for (int i = 0; i < numbers.Length - 1; i++)
+            {
+                for (int j = i + 1; j < numbers.Length; j++)
+                {
+                    if (numbers[i] > numbers[j])
+                    {
+                        temp = numbers[i];
```
Comment (westonpace) [inline FromZeroToHero.sln:32]: I'm not running this to verify (to save time on my end) but it looks like a valid implementation of a sort here.  As an optimization you could consider tracking if you had to move any numbers.  If you don't you can bail early.  This helps run more quickly when a list is mostly (or completely) sorted already.  Not required for the assignment of course.

File: FromZeroToHero.sln [hunk 5]
```diff
@@ -1,15 +1,177 @@
...
+                        numbers[i] = numbers[j];
+                        numbers[j] = temp;
+                    }
+                }
+            }
+            Console.WriteLine("Sorted array");
+            foreach (int number in numbers)
+            {
+
+                Console.Write(number + " ");
```
Comment (westonpace) [inline FromZeroToHero.sln:42]: This puts an extra space at the end of the line.  Also, you should probably have `Console.WriteLine` after the loop to put a newline.

File: FromZeroToHero.sln [hunk 6]
```diff
@@ -1,15 +1,177 @@
...
+            Console.WriteLine("Remove first element in array");
+            for (int i = 0; i < numbers.Length - 1; i++)
+            {
+                Console.Write(newArray[i] + " ");
+            }
+        }
+
+        private static void RemoveElementAtEnd(int[] numbers)
+        {
+            int[] newArray = new int[numbers.Length - 1];
```
Comment (westonpace) [inline FromZeroToHero.sln:133]: Very consistent and precise array methods here.  One thing you might consider is reusing your general purpose methods (e.g. `RemoveElementAtPosition`) to implement your special purpose methods (e.g. `RemoveElementAtEnd`).

File: FromZeroToHero.sln [hunk 7]
```diff
@@ -1,15 +1,177 @@
...
+                    newArray[i] = numbers[i];
+                }
+                if (i >= index)
+                {
+                    newArray[i] = numbers[i + 1];
+                }
+
+            }
+            Console.WriteLine();
+            Console.WriteLine("Remove element at any posiion in array");
```
Comment (westonpace) [inline FromZeroToHero.sln:161]: Nit: position

File: FromZeroToHero.sln [hunk 8]
```diff
@@ -1,15 +1,177 @@
...
+            for (int i = 0; i < numbers.Length - 1; i++)
+            {
+                Console.Write(newArray[i] + " ");
+            }
+        }
+        static void LoginInfo(string name, string password)
+        {
+            string myName = "Kin";
+            string pwd = "21Lome";
+            if (myName.Equals(name, StringComparison.OrdinalIgnoreCase) && pwd.Equals(password, StringComparison.OrdinalIgnoreCase))
```
Comment (westonpace) [inline FromZeroToHero.sln:171]: This is not correct.  According to the requirements passwords should not be compared with `OrdinalIgnoreCase`.

**Review by westonpace — COMMENTED** [review]
Good submission.  Your style is very consistent and it makes it easy to read.

File: FromZeroToHero.sln (hunk 1 above)
Comment (florencegogovor) [reply FromZeroToHero.sln:2]: It is my first time, sorry for  this confusion
I don't find the homework branch in the repo.
Where should I make the pull request for the future homework

**Comment by Almantask** [comment]
expired

**Pull request closed without merging**